its/SonarCloudIntegrationTests.cs

```diff
@@ -0,0 +1,192 @@
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.WebApi;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace AzureDevOpsExtension.IntegrationTests
+{
+    [TestClass]
+    public class SonarCloudIntegrationTests
+    {
+        public TestContext TestContext { get; set; }
+
+        private static SonarCloudCallWrapper _scInstance;
+        private static String _buildParameters;
+        private static BuildHttpClient _buildHttpClient;
+
+        [ClassInitialize]
+        public static void SetupTests(TestContext testContext)
+        {
+            _scInstance = new SonarCloudCallWrapper();
+            _buildParameters = "{\"IT_TRIGGER_BUILD\":\"true\"}";
+            VssConnection connection = GetAzDoConnection();
+
+            _buildHttpClient = connection.GetClient<BuildHttpClient>();
+        }
+
+        [TestMethod]
+        public async Task Execute_Scannercli_Build_And_Analysis()
```

Comment:
It looks like the four tests are identical except for four variable (pipeline to run, project key, expected LOC, expected coverage). 

Also, the configuration is quite complicated: the loc/coverage numbers are hard-coded, but the pipeline/project key are pipeline variables, passed through the yaml file and picked up by the tests as environment variables.

Could the duplication be removed and the setup simplified by making the tests data-driven? e.g.

`[TestMethod]`
`[DataRow("cli-pipeline", "cli-proj", 123, 456)]`
...

`public async Task Execute_Build_Analysis(string pipelineName, string projectKey, int expectedLoc, double expectedCoverage)`
...